pkg/types/aws/validation/machinepool.go

```diff
@@ -52,6 +113,40 @@ func ValidateMachinePool(platform *aws.Platform, p *aws.MachinePool, fldPath *fi
 	return allErrs
 }
 
+func validateVolumeSize(p *aws.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	volumeType := p.EC2RootVolume.Type
+	volumeSize := p.EC2RootVolume.Size
+
+	if volumeLimits, ok := validLimits[volumeType]; ok {
+		validSize := volumeLimits.VolumeSize
+		if volumeSize >= validSize.Max || volumeSize <= validSize.Min {
```

Comment:
So the Min, Max values themselves are not valid?

Reply:
No but the static checks are actually being changed. The max size isn't quite that simple. There is a max size per region per instance, so there is no way to actually check that in the types package.